### PR TITLE
Add Random Emoji on-chain NFT on OpenSea example

### PR DESCRIPTION
**Pull request by andrejrakic**
Added Random Emoji on-chain NFT on OpenSea example. This is a Hardhat with Typescript project which aims to demonstrate how to use Chainlink VRF to generate fully on-chain SVGs and host them on OpenSea on Rinkeby. The repo contains 
- main smart contract,
- mock oracle contracts for testing
- unit tests
- deployment script which also auto-fund contract with 1 Rinkeby LINK and verifies it on Etherscan

### CODE REVIEW

File: random-svg-nft/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+After that run the deployment script which will
+
+- deploy your smart contract to the Rinkeby
+- fund it with 1 Rinkeby LINK
+- verify it on Etherscan
+
+```shell
+yarn deploy
```
Comment (PatrickAlphaC) [inline random-svg-nft/README.md:68]:
```suggestion
yarn deploy --network rinkeby
```

File: random-svg-nft/contracts/EmojiNFT.sol [hunk 2]
```diff
@@ -0,0 +1,123 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/Base64.sol";
+
+contract EmojiNFT is ERC721URIStorage, VRFConsumerBase {
```
Comment (PatrickAlphaC) [inline random-svg-nft/contracts/EmojiNFT.sol:10]: Ahhh, sorry. We'll want to use VRF2 for this. Getting announced tomorrow. 

https://documentation-git-fork-dwightjl-dwightjl-vrf-v2-chainlinklabs.vercel.app/docs/chainlink-vrf/

File: random-svg-nft/.env.example [hunk 3]
```diff
@@ -0,0 +1,3 @@
+ETHERSCAN_API_KEY=ABC123ABC123ABC123ABC123ABC123ABC1
+RINKEBY_URL=https://eth-rinkeby.alchemyapi.io/v2/<YOUR ALCHEMY KEY>
+PRIVATE_KEY=0xabc123abc123abc123abc123abc123abc123abc123abc123abc123abc123abc1
```
Comment (rgottleber) [inline random-svg-nft/.env.example:3]:
```suggestion
ETHERSCAN_API_KEY=<YOUR ETHERSCAN API>
RINKEBY_URL=https://eth-rinkeby.alchemyapi.io/v2/<YOUR ALCHEMY KEY>
PRIVATE_KEY=<YOUR PRIVATE KEY>
```

**Review by rgottleber — COMMENTED** [review]
A couple of thoughts/questions. Looks great!

**Comment by PatrickAlphaC** [comment]
@rgottleber you need me to review? Or is @andrejrakic helping you?

File: random-svg-nft/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+1. Install packages
+
+```shell
+yarn install
+```
+
+2. Compile contracts
+
+```shell
+yarn compile
```
[the remark this answers is not in the document]
Comment (andrejrakic) [reply random-svg-nft/README.md:30]: These warnings come from the contracts from the `@chainlink/contracts` package actually, not sure we can do much to fix them. We can set the Logger level to log only errors, but that can hide some valuable info if someone wants to build on top of this project. Same warnings are also present in the hardhat-starter-kit, too. Thoughts on this?

File: random-svg-nft/.env.example (hunk 3 above)
Comment (andrejrakic) [reply random-svg-nft/.env.example:3]: It makes sense to me 👍🏻

**Comment by andrejrakic** [comment]
@rgottleber, can you review new changes? I've migrated the whole project to use VRF v2. Thanks

File: random-svg-nft/contracts/EmojiNFT.sol [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+  function fulfillRandomWords(uint256 requestId, uint256[] memory randomNumbers)
+    internal
+    override
+  {
+    uint256 tokenId = tokenIds.current();
+
+    uint256 emojiIndex = (randomNumbers[0] % emojis.length) + 1;
+    string memory emoji = emojis[emojiIndex];
+    string memory color = pickRandomColor(randomNumbers[1], randomNumbers[2], randomNumbers[3]);
```
Comment (rgottleber) [inline random-svg-nft/contracts/EmojiNFT.sol:127]: We are expecting 4 numbers here but above, line 44 we are allowing the constructor to determine the number of words, do we want to allow the ability to determine the number of words when we have to have at least 4?

Comment (andrejrakic) [reply random-svg-nft/contracts/EmojiNFT.sol:127]: Good catch!

**Review by rgottleber — APPROVED** [review]

**Pull request merged**